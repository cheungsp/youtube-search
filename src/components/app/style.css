body {
  background-color: whitesmoke;
}

.video-item img {
  max-width: 64px;
}

.video-detail .details {
  margin-top: 10px;
  padding: 10px;
  border: 1px solid #ddd;
  border-radius: 4px;
}