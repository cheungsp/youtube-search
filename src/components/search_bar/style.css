.search-bar {
  margin: 20px;
  text-align: center;
}

.search-bar input {
  width: 80%;
}

/*
"test": "jest"
"test": "react-scripts test --env=jsdom",
*/