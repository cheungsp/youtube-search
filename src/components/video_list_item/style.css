.single-video-item {
  margin-bottom: 15px;
}