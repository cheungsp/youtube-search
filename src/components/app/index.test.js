import React from 'react';
import { shallow, mount } from 'enzyme'
import App from './index.js'
import YTSearch from 'youtube-api-search';
import VideoList from '../video_list/index';
import VideoInfo from '../video_info/index';
import VideoListItem from '../video_list_item/index';
import SearchBar from '../search_bar/index';
import toJson from 'enzyme-to-json'
import './style.css';


test('App component should render as expected', () => {
  const component = shallow(<App />)
  const tree =  toJson(component)
  expect(tree).toMatchSnapshot()
})

describe('<App />', () => {
  describe('it renders app with correct state', () => {
    const component = mount(<App />);
    console.log(component)
    expect(component.node.state.videos.length).toBe(0)
    expect(component.node.state.selectedVideos).toBe(undefined)
  })
})

