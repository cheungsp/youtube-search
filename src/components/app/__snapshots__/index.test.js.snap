// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render as expected 1`] = `
<div>
  <SearchBar
    onSearchTermChange={[Function]}
  />
  <VideoInfo
    video={null}
  />
  <VideoList
    onVideoSelect={[Function]}
    videos={Array []}
  />
</div>
`;
